llms.md

```diff
@@ -0,0 +1,1039 @@
+        }
+    }
+
+    // Example: Filtering and Deleting with Indexes
+    [Reducer]
+    public static void IndexOperations(ReducerContext ctx)
+    {
+        // Example: Find items named "Sword"
+        var items = ctx.Db.items.name.Filter("Sword");
+        foreach (var item in items) { /* ... */ }
+
+        // Example: Delete points where x is 5
+        bool deleted = ctx.Db.points.idx_xy.Delete(5L); // Filter on index prefix
+
+        // Example: Find points where x = 3 and y = 7
+        var specificPoint = ctx.Db.points.idx_xy.Filter((3L, 7L));
+        foreach(var pt in specificPoint) { /* ... should be at most one */}
+
+        // Example: Find points where x is between 100 and 200 (inclusive)
+        var pointsInRange = ctx.Db.points.idx_xy.Filter((100L, 200L));
+        foreach(var pt in pointsInRange) { /* ... */ }
+
+        // Example: Delete items named "Shield"
+        uint numDeleted = ctx.Db.items.name.Delete("Shield");
+        Log.Info($"Deleted {numDeleted} Shield(s)");
+    }
+}
+```
+
+:::note C# `Insert` vs Rust `try_insert`
+Unlike Rust, the C# SDK does not currently provide a `TryInsert` method that returns a result. The standard `Insert` method will throw an exception if a constraint (primary key, unique index) is violated. Therefore, C# reducers should typically check for potential constraint violations *before* calling `Insert`, or be prepared to handle the exception (which will likely roll back the transaction).
+:::
+
+##### Module Identity vs Sender Identity
+
+Inside a reducer, the `ReducerContext` provides two important identities:
+
+*   `ctx.Sender`: The [`Identity`](#identity) of the client who called the reducer, or the identity of the module itself if the reducer was called by the scheduler (for scheduled reducers) or internally by another reducer.
+*   `ctx.Identity`: The [`Identity`](#identity) of the module (database) itself.
+*   `ctx.Timestamp`: A [`Timestamp`](#timestamp) indicating when the reducer execution began.
+*   `ctx.ConnectionId`: A [`ConnectionId`](#connectionid) representing the specific connection that invoked the reducer. This is a large (u128-based), randomly generated identifier **assigned by the SpacetimeDB server** for each distinct client connection (e.g., a WebSocket session or a stateless HTTP call). It is unique for the duration of that connection instance (from `client_connected` to `client_disconnected`). Unlike `ctx.Sender` (the authenticated [`Identity`](#identity)), the `ConnectionId` is **not verified** based on client input; it solely identifies the server-side connection object. Use `ConnectionId` for logic specific to a *transient connection* (e.g., tracking session state, rate limiting) and `Sender` (`Identity`) for logic tied to the *persistent, authenticated user account*.
+
+This distinction is crucial for security, especially with scheduled reducers. You often want to ensure that only the scheduler (i.e., the module itself) can trigger certain actions.
```

Comment:
This is a very weird section. Either the framing is wrong, or `ctx.Timestamp` and `ctx.ConnectionId` do not belong here.

Reply:
Ah I think the vibes were off here lol, I'll fix it up